lib/wallet/src/Cardano/Wallet/DB/Store/Submissions/New/Operations.hs

```diff
@@ -0,0 +1,142 @@
+
+
+syncSubmissions :: WalletId -> TxSubmissions -> TxSubmissions -> SqlPersistT IO ()
+syncSubmissions wid old new = do
+
+    let deletes = transactions old `Map.difference` transactions new
+    forM_ (Map.keys deletes) $ \k -> delete (SubmissionsKey k)
+
+    let repserts = transactions new
+    forM_ (Map.assocs repserts) $ \(iden, status) -> do
+        let result = case status of
+                Sbm.Expired expiring (_, sealed)
+                    -> Just (sealed, expiring, Nothing, ExpiredE)
+                Sbm.InSubmission expiring (_, sealed)
+                    -> Just (sealed, expiring,Nothing, InSubmissionE)
+                Sbm.InLedger expiring acceptance (_, sealed)
+                    -> Just (sealed, expiring, Just acceptance, InLedgerE)
+                Sbm.Unknown -> Nothing
+        case result of
+            Just (sealed, expiring, acceptance,statusNumber ) -> repsert
+                (SubmissionsKey iden)
+                (Submissions iden sealed expiring acceptance wid statusNumber)
+            Nothing -> pure ()
+    repsert
+        (SubmissionsSlotsKey wid)
+        $ SubmissionsSlots (finality new) (tip new) wid
+
+instance Sbm.HasTxId (TxId, W.SealedTx) where
+    type TxId (TxId, W.SealedTx) = TxId
+    txId (iden,_) = iden
+
+data SubmissionsError
+    = SubmissionsSlotsMissingForWallet WalletId
+    | MoreThanOneSubmissionsSlotsDefinedForWallet WalletId
+    deriving (Show, Eq, Exception)
+
+mkStoreAnySubmissions
+    :: (Base d ~ TxSubmissions, Delta d)
+    => WalletId
+    -> Store (SqlPersistT IO) d
+mkStoreAnySubmissions wid =
+    Store
+    { loadS = do
+        slots <- selectList [SubmissionsSlotsWallet ==. wid] []
+        txs <- selectList [SubmissionWallet ==. wid ] []
+        pure $ case slots of
+            [] -> Left $ SomeException $ SubmissionsSlotsMissingForWallet wid
+            [Entity _ (SubmissionsSlots finality' tip' _)] -> Right
+                $ Sbm.Submissions (mkTransactions tip' txs) finality' tip'
+            _ -> Left $ SomeException
+                    $ MoreThanOneSubmissionsSlotsDefinedForWallet wid
+    , writeS = syncSubmissions wid (Sbm.Submissions mempty 0 0)
+    , updateS = \base delta -> syncSubmissions wid base $ apply delta base
+    }
+
+mkTransactions :: SlotNo -> [Entity Submissions] -> Map TxId TxSubmissionsStatus
+mkTransactions tip' xs = Map.fromList $ do
+    Entity _  (Submissions iden sealed expiration acceptance _ status) <- xs
+    pure (iden, mkStatus tip' iden sealed expiration acceptance status)
+
+mkStatus :: SlotNo -> TxId -> W.SealedTx -> SlotNo -> Maybe SlotNo
+    -> TxSubmissionStatusEnum -> TxSubmissionsStatus
+mkStatus _ iden sealed expiring (Just acceptance) InLedgerE
+    = Sbm.InLedger expiring acceptance (iden, sealed)
+mkStatus tip' iden sealed expiring Nothing InSubmissionE
+    | expiring > tip' = Sbm.InSubmission expiring (iden, sealed)
+    | otherwise = Sbm.Unknown
```

Comment:
I think that at this point, it should not be necessary to check that the data in the database satisfies any invariants — a transaction which is `InSubmission` and whose expiry slot is after the tip should not make it into the database in the first place. 🤔 In fact, 

(Using `Sbm.Unknown` here has the small drawback that it is stored in the `_transactionsL` `Map` — but transactions with `Unknown` status are typically not even present in that map.)

Reply:
I have decided to drop the tests for the invariants, but also came to the conclusion that the work required to distinguish bad data from good data would have too little value here to be doing at this moment. I have added a note in a comment.